zrml/orderbook/src/tests.rs

```diff
@@ -107,367 +541,506 @@ fn it_places_orders() {
             RuntimeOrigin::signed(BOB),
             market_id,
-            Asset::CategoricalOutcome(0, 1),
-            OrderSide::Ask,
-            10,
-            5,
+            maker_asset,
+            maker_amount,
+            market.base_asset,
+            taker_amount,
         ));
 
-        let shares_reserved = Tokens::reserved_balance(Asset::CategoricalOutcome(0, 1), &BOB);
-        assert_eq!(shares_reserved, 10);
+        let shares_reserved = AssetManager::reserved_balance(maker_asset, &BOB);
+        assert_eq!(shares_reserved, maker_amount);
     });
 }
 
 #[test]
-fn it_fills_ask_orders_fully() {
+fn it_fills_order_fully_maker_outcome_asset() {
     ExtBuilder::default().build().execute_with(|| {
         let market_id = 0u128;
         let market = market_mock::<Runtime>();
-        Markets::<Runtime>::insert(market_id, market);
+        Markets::<Runtime>::insert(market_id, market.clone());
+
+        let maker_asset = Asset::CategoricalOutcome(0, 1);
+        let taker_asset = market.base_asset;
 
-        let outcome_asset = Asset::CategoricalOutcome(0, 1);
-        // Give some shares for Bob.
-        assert_ok!(Tokens::deposit(outcome_asset, &BOB, 100));
+        let maker_amount = 100 * BASE;
+        let taker_amount = 500 * BASE;
+        assert_ok!(AssetManager::deposit(maker_asset, &BOB, maker_amount));
 
-        // Make an order from Bob to sell shares.
         assert_ok!(Orderbook::place_order(
             RuntimeOrigin::signed(BOB),
             market_id,
-            outcome_asset,
-            OrderSide::Ask,
-            10,
-            50,
+            maker_asset,
+            maker_amount,
+            taker_asset,
+            taker_amount,
         ));
 
-        let reserved_bob = Tokens::reserved_balance(outcome_asset, &BOB);
-        assert_eq!(reserved_bob, 10);
+        let reserved_bob = AssetManager::reserved_balance(maker_asset, &BOB);
+        assert_eq!(reserved_bob, maker_amount);
+
+        let market_creator_balance_before =
+            AssetManager::free_balance(taker_asset, &MARKET_CREATOR);
 
         let order_id = 0u128;
+        assert_ok!(AssetManager::deposit(taker_asset, &ALICE, taker_amount));
         assert_ok!(Orderbook::fill_order(RuntimeOrigin::signed(ALICE), order_id, None));
 
-        let reserved_bob = Tokens::reserved_balance(outcome_asset, &BOB);
+        let market_creator_balance_after = AssetManager::free_balance(taker_asset, &MARKET_CREATOR);
+        let taker_fees = ExternalFees::<Runtime, FeeAccount>::get_fee(market_id, taker_amount);
+        assert_eq!(market_creator_balance_after - market_creator_balance_before, taker_fees);
+
+        let reserved_bob = AssetManager::reserved_balance(maker_asset, &BOB);
         assert_eq!(reserved_bob, 0);
 
         System::assert_last_event(
             Event::<Runtime>::OrderFilled {
                 order_id,
                 maker: BOB,
                 taker: ALICE,
```

Comment:
This event doesn't include which part of the maker amount was traded. I guess we could read that off the the transaction data. Is that standard practice, or should we add the other amounts too, @saboonikhil?